doc/guides/dts/index.rst

```diff
@@ -398,6 +398,91 @@ usually take a ``struct device*`` as their first argument. This allows the
 driver API to use information from devicetree to interact with the device
 hardware.
 
+Generated macro rules
+=====================
+.. code-block:: none
+
+      aliases {
+           i2c-0 = &i2c0;
+      };
+
+      soc {
+           i2c0: i2c@40066000 {
+                compatible = "nxp,kinetis-i2c";
+                reg = <0x40066000 0x1000>;
+                fxos8700@1d {
+                     compatible = "nxp,fxos8700";
+                     reg = <0x1d>;
+                };
+           };
+      };
+
+.. list-table::
+   :header-rows: 1
+
+   * - type
```

Comment:
Could capitalize the headings, etc. Nice to aim for more "writing class" than "engineering class" in the documentation.